src/google/cloud/ndb/key.py

```diff
@@ -837,14 +837,30 @@ def get_async(
 
         cls = model.Model._kind_map.get(self.kind())
 
+        if cls:
+            cls._pre_get_hook(self)
+
         @tasklets.tasklet
         def get():
-            if cls:
-                cls._pre_get_hook(self)
+            if _options.use_cache:
+                try:
+                    # This result may be None even on a cache hit.
```

Comment:
"even" implies that it could also return None if it's not a cache hit, but all cache misses raise KeyError.

Maybe:

~~~~
# This result may be None, if None is cached for this key
~~~~

Or similar?

Reply:
Done, thanks for the suggestion